hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/DietTransformations.scala

```diff
@@ -0,0 +1,227 @@
+      dfDailySupplementsEnzyme = rawRecord.getOptionalNumber("df_s_enzyme").filter(_ != 0L),
+      dfDailySupplementsProbiotics = rawRecord.getOptionalNumber("df_s_probiotics").filter(_ != 0L),
+      dfDailySupplementsFiber = rawRecord.getOptionalNumber("df_s_fiber").filter(_ != 0L),
+      dfDailySupplementsAlkalinize = rawRecord.getOptionalNumber("df_s_alkalinize").filter(_ != 0L),
+      dfDailySupplementsAcidify = rawRecord.getOptionalNumber("df_s_acidify").filter(_ != 0L),
+      dfDailySupplementsTaurine = rawRecord.getOptionalNumber("df_s_taurine").filter(_ != 0L),
+      dfDailySupplementsAntiox = rawRecord.getOptionalNumber("df_s_antiox").filter(_ != 0L),
+      dfDailySupplementsCoenzymeQ10 = rawRecord.getOptionalNumber("df_s_q10"),
+      dfDailySupplementsOther = otherSupps,
+      dfDailySupplementsOtherDescription = if (otherSupps.nonEmpty) {
+        rawRecord.getOptional("df_s_other_text")
+      } else {
+        None
+      }
+    )
+  }
+
+  /** Map fields about supplements taken less-than-daily by a dog. */
+  def mapInfrequentSupplements(rawRecord: RawRecord, dog: HlesDogDiet): HlesDogDiet = {
+    val otherSupps = rawRecord.getOptionalNumber("df_s_other_ltd").filter(_ != 0L)
+
+    dog.copy(
+      dfInfrequentSupplementsBoneMeal =
+        rawRecord.getOptionalNumber("df_s_bone_meal_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsGlucosamine =
+        rawRecord.getOptionalNumber("df_s_glucosamine_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsChondroitin =
+        rawRecord.getOptionalNumber("df_s_chondroitin_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsOtherJoint =
+        rawRecord.getOptionalNumber("df_s_joint_other_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsOmega3 =
+        rawRecord.getOptionalNumber("df_s_omega3_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsNonOilSkin =
+        rawRecord.getOptionalNumber("df_s_skin_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsVitamins =
+        rawRecord.getOptionalNumber("df_s_vitamin_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsEnzyme =
+        rawRecord.getOptionalNumber("df_s_enzyme_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsProbiotics =
+        rawRecord.getOptionalNumber("df_s_probiotics_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsFiber = rawRecord.getOptionalNumber("df_s_fiber_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsAlkalinize =
+        rawRecord.getOptionalNumber("df_s_alkalinize_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsAcidify =
+        rawRecord.getOptionalNumber("df_s_acidify_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsTaurine =
+        rawRecord.getOptionalNumber("df_s_taurine_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsAntiox =
+        rawRecord.getOptionalNumber("df_s_antiox_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsCoenzymeQ10 = rawRecord.getOptionalNumber("df_s_q10_ltd"),
+      dfInfrequentSupplementsOther = otherSupps,
```

Comment:
Should 0 be filtered out of the "other" and "Coenzyme Q10" fields too?

Reply:
It's filtered out of "other" at the `val` declaration, but good catch on the Q10